src/platform/datapath_raw_xdp_win.c

```diff
@@ -107,246 +108,137 @@ CxPlatXdpExecute(
-
-            IUnknown** rawArray;
-            SafeArrayAccessData(pSafeArray, (void**)&rawArray);
-
-            // Set up the RSS table according to number of procs and proc groups.
-            numberOfProcs = GetActiveProcessorCount(ALL_PROCESSOR_GROUPS);
-            numberOfProcGroups = GetActiveProcessorGroupCount();
-            rssTableSize = numberOfProcs * numberOfProcGroups;
-            rssTable = malloc(rssTableSize);
-            memset(rssTable, 0, rssTableSize);
-
-            for (long i = lLower; i <= lUpper; i++)
-            {
-                IUnknown* pIUnk = rawArray[i];
-                IWbemClassObject *obj = NULL;
-                pIUnk->lpVtbl->QueryInterface(pIUnk, &IID_IWbemClassObject, (void **)&obj);
-                if (obj == NULL) {
-                    QuicTraceEvent(
-                        LibraryErrorStatus,
-                        "[ lib] ERROR, %u, %s.",
-                        hRes,
-                        "QueryInterface");
-                    free(rssTable);
-                    hRes = QUIC_STATUS_OUT_OF_MEMORY;
-                    goto Cleanup;
-                }
+        XSK_RING_INFO_SET TxRingInfo;
+        uint32_t TxRingInfoSize = sizeof(TxRingInfo);
+        Status = Xdp->XdpApi->XskGetSockopt(TxXsk, XSK_SOCKOPT_RING_INFO, &TxRingInfo, &TxRingInfoSize);
+        if (QUIC_FAILED(Status)) { CloseHandle(TxXsk); return Status; }
 
-                hr = obj->lpVtbl->Get(obj, L"ProcessorNumber", 0, &vtProp, 0, 0);
-                UINT32 procNum = vtProp.iVal;
-                VariantClear(&vtProp);
-                hr = obj->lpVtbl->Get(obj, L"ProcessorGroup", 0, &vtProp, 0, 0);
-                UINT32 groupNum = vtProp.iVal;
-                VariantClear(&vtProp);
-                CXPLAT_DBG_ASSERT(groupNum < numberOfProcGroups);
-                CXPLAT_DBG_ASSERT(procNum < numberOfProcs);
-                *(rssTable + groupNum * numberOfProcs + procNum) = 1;
-                obj->lpVtbl->Release(obj);
-            }
+        XSK_RING TxRing, TxCompletionRing;
+        XskRingInitialize(&TxRing, &TxRingInfo.Tx);
+        XskRingInitialize(&TxCompletionRing, &TxRingInfo.Completion);
 
-            SafeArrayUnaccessData(pSafeArray);
+        uint32_t TxIndex;
+        XskRingProducerReserve(&TxRing, MAXUINT32, &TxIndex);
 
-            // Count unique RSS procs by counting ones in rssTable.
-            for (DWORD i = 0; i < rssTableSize; ++i) {
-                cnt += rssTable[i];
-            }
+        XSK_BUFFER_DESCRIPTOR* Buffer = XskRingGetElement(&TxRing, TxIndex++);
+        Buffer->Address.BaseAddress = 0;
+        Buffer->Address.Offset = FIELD_OFFSET(XDP_TX_PACKET, FrameBuffer);
+        Buffer->Length = TxPacket.Buffer.Length;
+        XskRingProducerSubmit(&TxRing, 1);
 
-            free(rssTable);
-        }
+        XSK_NOTIFY_RESULT_FLAGS OutFlags;
+        Status = Xdp->XdpApi->XskNotifySocket(TxXsk, XSK_NOTIFY_FLAG_POKE_TX|XSK_NOTIFY_FLAG_WAIT_TX, 1000, &OutFlags);
```

Comment:
1000 magic number - should be `#define` ?

Reply:
Will do.